weather.py

```diff
@@ -0,0 +1,306 @@
+# Imports written in alphabetical order
+from os import path
+import sys
+from termcolor import colored
+import zipfile
+
+
+class Weatherman:  # Weatherman Application in one class
+
+    file_pre = "Murree_weather_"  # The part of the file name that is same for all files
+
+    # Dictionaries as class instances to store calculation results
+    e_result = dict()
+    a_result = dict()
+    c_result = dict()
+
+    # Report Number to print
+    report_num = 1
+
+    # Initialize Lists in these indexes of dictionary
+    c_result["max_temp"] = []
+    c_result["min_temp"] = []
+
+    # List with months to access by numerical form
+    months = ["Jan", "Feb", "Mar",
+              "Apr", "May", "Jun",
+              "Jul", "Aug", "Sep",
+              "Oct", "Nov", "Dec"
+              ]
+
+    # Class Constructor
+    def __init__(self, path, e, a, c, b):
+        self.__path = path
+        self.__e = e
+        self.__a = a
+        self.__c = c
+        self.__b = b
+
+        # Variables used to cater multiple inputs of a specific argument
+        self.__curr_e = None
+        self.__curr_a = None
+        self.__curr_c = None
+
+    # Method to extract weather files to specified path
+    def Extract(self):
```

Comment:
In python, methods names don't contain capital letters for further styling rules please consult [PEP 8](https://peps.python.org/pep-0008/).
`extract` also doesn't seem the part of `Weatherman` class you can extract it as a utility method.
```suggestion
    def extract_weather_files(self):
```

Reply:
change made as suggested.